File: src/main/kotlin/fr/godox/customjetbrainspluginrepo/config/ClientWebConfig.java
package fr.godox.customjetbrainspluginrepo.config;

import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.transaction.annotation.EnableTransactionManagement;
import org.springframework.web.servlet.config.annotation.EnableWebMvc;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;

// On ajoute EnableTransactionManagement
@EnableWebMvc
@Configuration
@ComponentScan(basePackages = {"fr.godox.customjetbrainspluginrepo.controller", "fr.godox.customjetbrainspluginrepo.service"})
@EnableTransactionManagement
public class ClientWebConfig implements WebMvcConfigurer {

    public ClientWebConfig() {
        try {
            Files.createDirectories(new File("./received").toPath());
        } catch (IOException ignored) {

        }
    }

    //    @Bean
//    @Description("Thymeleaf Template Resolver")
//    public SpringResourceTemplateResolver templateResolver() {
//        SpringResourceTemplateResolver templateResolver = new SpringResourceTemplateResolver();
//        templateResolver.setPrefix("/WEB-INF/views/");
//        templateResolver.setSuffix(".html");
//        templateResolver.setTemplateMode("HTML5");
//        return templateResolver;
//    }
//
//    @Bean
//    @Description("Thymeleaf Template Engine")
//    public SpringTemplateEngine templateEngine() {
//        SpringTemplateEngine templateEngine = new SpringTemplateEngine();
//        templateEngine.addDialect(new Java8TimeDialect());
//        templateEngine.setTemplateResolver(templateResolver());
//        //templateEngine.setTemplateEngineMessageSource(messageSource());
//        return templateEngine;
//    }
//
//
//
//
//    @Bean
//    @Description("Thymeleaf View Resolver")
//    public ThymeleafViewResolver viewResolver() {
//        ThymeleafViewResolver viewResolver = new ThymeleafViewResolver();
//        viewResolver.setTemplateEngine(templateEngine());
//        viewResolver.setOrder(1);
//        return viewResolver;
//    }

    /*
    @Bean
    @Description("Spring Message Resolver")
    public ResourceBundleMessageSource messageSource() {
        ResourceBundleMessageSource messageSource = new ResourceBundleMessageSource();
        messageSource.setBasename("messages");
        return messageSource;
    }


     */


//    @Bean
//    public MappingJackson2XmlHttpMessageConverter mappingJackson2XmlHttpMessageConverter(Jackson2ObjectMapperBuilder builder) {
//        XmlMapper xmlMapper = builder.createXmlMapper(true).build();
//        xmlMapper.configure(ToXmlGenerator.Feature.WRITE_XML_DECLARATION, true);
//        return new MappingJackson2XmlHttpMessageConverter(xmlMapper);
//    }

//    @Bean
//    public DataSource dataSource() {
//        EmbeddedDatabaseBuilder builder = new EmbeddedDatabaseBuilder();
//        return builder
//                .setType(EmbeddedDatabaseType.H2) //.H2 or .DERBY
////                .addScript("db/sql/create-db.sql")   // dans resources
//                .build();
//    }

//    @Bean
//    public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
//        LocalContainerEntityManagerFactoryBean em = new LocalContainerEntityManagerFactoryBean();
//        em.setDataSource(dataSource());
//        em.setPackagesToScan("fr.godox.customjetbrainspluginrepo.model");
//
//        JpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
//        em.setJpaVendorAdapter(vendorAdapter);
//        em.setJpaProperties(additionalProperties());
//
//        return em;
//    }

//    Properties additionalProperties() {
//        Properties properties = new Properties();
//        properties.setProperty("hibernate.hbm2ddl.auto", "create-drop");
//        properties.setProperty(
//                "hibernate.dialect", "org.hibernate.dialect.H2Dialect");
//        properties.setProperty("hibernate.hbm2ddl.import_files", "insert-data.sql");
//        return properties;
//    }

//    @Bean
//    public PlatformTransactionManager transactionManager(
//            EntityManagerFactory emf) {
//        JpaTransactionManager transactionManager = new JpaTransactionManager();
//        transactionManager.setEntityManagerFactory(emf);
//
//        return transactionManager;
//    }


}

File: src/main/kotlin/fr/godox/customjetbrainspluginrepo/config/ClientWebConfig.kt
package fr.godox.customjetbrainspluginrepo.config

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.PropertyName
import com.fasterxml.jackson.databind.introspect.AnnotatedClass
import com.fasterxml.jackson.databind.introspect.AnnotatedMember
import com.fasterxml.jackson.databind.introspect.JacksonAnnotationIntrospector
import com.fasterxml.jackson.dataformat.xml.XmlMapper
import fr.godox.customjetbrainspluginrepo.repository.DescriptorsTable
import fr.godox.customjetbrainspluginrepo.repository.IdeaVersion
import fr.godox.customjetbrainspluginrepo.repository.PluginVendor
import org.jetbrains.exposed.sql.Database
import org.jetbrains.exposed.sql.SchemaUtils
import org.jetbrains.exposed.sql.transactions.transaction
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.core.env.Environment
import org.springframework.http.converter.xml.MappingJackson2XmlHttpMessageConverter
import org.springframework.transaction.annotation.EnableTransactionManagement
import org.springframework.web.servlet.config.annotation.EnableWebMvc
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
import java.io.File
import java.io.IOException
import java.io.Serializable
import java.nio.file.Files


@EnableWebMvc
@Configuration
@ComponentScan(basePackages = ["fr.godox.customjetbrainspluginrepo.controller", "fr.godox.customjetbrainspluginrepo.service"])
@EnableTransactionManagement
class ClientWebConfig(env: Environment) : WebMvcConfigurer {
    init {
        try {
            Files.createDirectories(File("./received").toPath())
            port = env.getProperty("server.port", Int::class.java)
            host = env.getProperty("server.address") ?: "localhost"
        } catch (ignored: IOException) {
        }


        Database.connect("jdbc:sqlite:repository.db", "org.sqlite.JDBC")

        transaction { SchemaUtils.create(DescriptorsTable, IdeaVersion, PluginVendor) }

    }

    class IgnoreInheritedIntrospector : JacksonAnnotationIntrospector() {
        override fun findRootName(ac: AnnotatedClass?): PropertyName {
            return super.findRootName(ac)
        }



        override fun hasIgnoreMarker(m: AnnotatedMember): Boolean {
            return !m.declaringClass.interfaces.contains(Serializable::class.java) || super.hasIgnoreMarker(m)
        }
    }

//    @Bean
//    fun xmlMapper(): MappingJackson2XmlHttpMessageConverter {
//        return MappingJackson2XmlHttpMessageConverter(XmlMapper().apply {
//            setAnnotationIntrospector(IgnoreInheritedIntrospector())
//        })
//    }

    @Bean
    @Primary
    @Suppress("SpringJavaInjectionPointsAutowiringInspection")
    fun objectMapper(converter: MappingJackson2XmlHttpMessageConverter): ObjectMapper {
        return converter.objectMapper.also {
            it.setAnnotationIntrospector(IgnoreInheritedIntrospector())
        }
    }


//
//    @Bean
//    fun initMapper(): Jackson2ObjectMapperBuilder {
//        return Jackson2ObjectMapperBuilder().apply {
//            filters(SimpleFilterProvider().setDefaultFilter(SimpleBeanPropertyFilter.filterOutAllExcept()))
//            annotationIntrospector(IgnoreInheritedIntrospector())
//            createXmlMapper(true)
//        }
//    }

    companion object {
        lateinit var host: String
        var port: Int? = null
    }

}
